### ISSUE TITLE
Topic subscribe count `topicCountForSequenceId` may contain dirty data

### ISSUE THREAD

**Issue by Technoboy-** [streamnative/mop#380]
**Describe the bug**
The below snippet is in MQTTProxyProtocolMethodProcessor#processSubscribe :
```
topicListFuture.thenCompose(topics -> {
            List<CompletableFuture<Void>> futures = new ArrayList<>();
            for (String topic : topics) {
                CompletableFuture<InetSocketAddress> lookupResult = lookupHandler.findBroker(TopicName.get(topic));
                futures.add(lookupResult.thenAccept(brokerAddress -> {
                    increaseSubscribeTopicsCount(msg.variableHeader().messageId(), 1);
                    writeToMqttBroker(channel, msg, topic, brokerAddress);
                }));
            }
            return FutureUtil.waitForAll(futures);
        }).exceptionally(ex -> {
            log.error("[Proxy Subscribe] Failed to process subscribe for {}", clientId, ex);
            int messageId = msg.variableHeader().messageId();
            MqttMessage subAckMessage = MqttUtils.isMqtt5(protocolVersion) ? MqttSubAckMessageHelper.createMqtt5(
                    messageId, Mqtt5SubReasonCode.UNSPECIFIED_ERROR, ex.getCause().getMessage()) :
                    MqttSubAckMessageHelper.createMqtt(messageId, Mqtt3SubReasonCode.FAILURE);
            channel.writeAndFlush(subAckMessage);
            channel.close();
            return null;
        });
```
If `writeToMqttBroker` occurs exception, `increaseSubscribeTopicsCount` may exist dirty data.

**To Reproduce**

**Expected behavior**
No dirty data existed.

**Screenshots**

**Desktop (please complete the following information):**

**Additional context**

**Comment by mattisonchao** [comment]
@Technoboy- 

Fixed by #381

**Comment by Technoboy-** [comment]
> MQTTProxyProtocolMethodProcessor

Only keep the order. We need to process the exception case in `exceptionally`
